packages/ng-query/src/lib/query.ts

```diff
@@ -106,149 +105,11 @@ export class QueryService {
-    options?: Omit<
-      NgQueryObserverOptions<
-        TQueryFnData,
-        TError,
-        TData,
-        TQueryFnData,
-        TQueryKey
-      >,
-      'queryKey' | 'initialData'
-    > & { initialData: TQueryFnData | (() => TQueryFnData) }
-  ): NgQueryObserverDefinedReturnType<TQueryFnData, TError, TData, TQueryKey>;
-
-  use<
-    TQueryFnData = unknown,
-    TError = unknown,
-    TData = TQueryFnData,
-    TQueryKey extends QueryKey = QueryKey
-  >(
-    queryKey: TQueryKey,
-    options?: Omit<
-      NgQueryObserverOptions<
-        TQueryFnData,
-        TError,
-        TData,
-        TQueryFnData,
-        TQueryKey
-      >,
-      'queryKey'
-    >
-  ): NgQueryObserverReturnType<TQueryFnData, TError, TData, TQueryKey>;
-
-  use<
-    TQueryFnData = unknown,
-    TError = unknown,
-    TData = TQueryFnData,
-    TQueryKey extends QueryKey = QueryKey
-  >(
-    queryKey: TQueryKey,
-    queryFn: ObservableQueryFn<TQueryFnData, TQueryKey>,
-    options?: Omit<
-      NgQueryObserverOptions<
-        TQueryFnData,
-        TError,
-        TData,
-        TQueryFnData,
-        TQueryKey
-      >,
-      'queryKey' | 'queryFn' | 'initialData'
-    > & { initialData?: () => undefined }
-  ): NgQueryObserverReturnType<TQueryFnData, TError, TData, TQueryKey>;
-
-  use<
-    TQueryFnData = unknown,
-    TError = unknown,
-    TData = TQueryFnData,
-    TQueryKey extends QueryKey = QueryKey
-  >(
-    queryKey: TQueryKey,
-    queryFn: ObservableQueryFn<TQueryFnData, TQueryKey>,
-    options?: Omit<
-      NgQueryObserverOptions<
-        TQueryFnData,
-        TError,
-        TData,
-        TQueryFnData,
-        TQueryKey
-      >,
-      'queryKey' | 'queryFn'
-    >
-  ): NgQueryObserverReturnType<TQueryFnData, TError, TData, TQueryKey>;
-
-  use<
-    TQueryFnData,
-    TError,
-    TData = TQueryFnData,
-    TQueryKey extends QueryKey = QueryKey
-  >(
-    arg1:
-      | TQueryKey
-      | NgQueryObserverOptions<
-          TQueryFnData,
-          TError,
-          TData,
-          TQueryFnData,
-          TQueryKey
-        >,
-    arg2?:
-      | ObservableQueryFn<TQueryFnData, TQueryKey>
-      | NgQueryObserverOptions<
-          TQueryFnData,
-          TError,
-          TData,
-          TQueryFnData,
-          TQueryKey
-        >,
-    arg3?: NgQueryObserverOptions<
-      TQueryFnData,
-      TError,
-      TData,
-      TQueryFnData,
-      TQueryKey
-    >
-  ):
-    | NgQueryObserverReturnType<TQueryFnData, TError, TData, TQueryKey>
-    | NgQueryObserverDefinedReturnType<TQueryFnData, TError, TData, TQueryKey> {
-    const parsedOptions = parseQueryArgs(
-      arg1,
-      arg2 as any,
-      arg3
-    ) as QueryOptions;
-
+  ): NgQueryObserverReturnType<TQueryFnData, TError, TData, TQueryKey> | NgQueryObserverDefinedReturnType<TQueryFnData, TError, TData, TQueryKey> {
     return buildQuery(
       this.instance,
       QueryObserver,
-      parsedOptions
+      options as QueryOptions
```

Comment:
Same for this conversion, i'd say this one is not as bad as the infinite query one but still what would you say?

Reply:
See https://github.com/TanStack/query/blob/v5.4.2/packages/svelte-query/src/createInfiniteQuery.ts

Reply:
Yes, we should copy the same overload and only change the queryFn and return data